cpp/src/IceGrid/ServerCache.cpp

```diff
@@ -792,37 +790,37 @@ ServerEntry::waitImpl(int timeout)
 void
 ServerEntry::synchronized()
 {
-    vector<SynchronizationCallbackPtr> callbacks;
+    vector<shared_ptr<SynchronizationCallback>> callbacks;
     {
-        Lock sync(*this);
+        lock_guard lock(_mutex);
         _callbacks.swap(callbacks);
     }
-    for(vector<SynchronizationCallbackPtr>::const_iterator p = callbacks.begin(); p != callbacks.end(); ++p)
+    for(const auto& callback : callbacks)
     {
         try
         {
-            (*p)->synchronized();
+            callback->synchronized();
         }
-        catch(...)
+        catch(const std::exception&)
```

Comment:
Why not keep `catch(...)` like below? or change below as well?

Reply:
I think both should be `catch(const std::exception&)`